test/optimizer/analyze_test.cpp

```diff
@@ -0,0 +1,118 @@
+    txn_manager_ = db_main_->GetTransactionLayer()->GetTransactionManager();
+
+    auto col = catalog::Schema::Column(
+        "attribute", type::TypeId::INTEGER, false,
+        parser::ConstantValueExpression(type::TransientValueFactory::GetNull(type::TypeId::INTEGER)));
+    StorageTestUtil::ForceOid(&(col), catalog::col_oid_t(1));
+    table_schema_ = catalog::Schema({col});
+    sql_table_ = new storage::SqlTable(db_main_->GetStorageLayer()->GetBlockStore().Get(), table_schema_);
+    tuple_initializer_ = sql_table_->InitializerForProjectedRow({catalog::col_oid_t(1)});
+
+    std::vector<catalog::IndexSchema::Column> keycols;
+    keycols.emplace_back("", type::TypeId::INTEGER, false,
+                         parser::ColumnValueExpression(CatalogTestUtil::TEST_DB_OID, CatalogTestUtil::TEST_TABLE_OID,
+                                                       catalog::col_oid_t(1)));
+    StorageTestUtil::ForceOid(&(keycols[0]), catalog::indexkeycol_oid_t(1));
+  }
+  void TearDown() override {
+    db_main_->GetTransactionLayer()->GetDeferredActionManager()->RegisterDeferredAction([=]() { delete sql_table_; });
+  }
+};
+
+/**
+ * This test creates fills in the table with i values for all i between 0 to 99. Then it runs the analyze function
+ and confirms the column stats generated.
+ */
+// NOLINTNEXTLINE
+TEST_F(AnalyzeTests, SingleColumnTest) {
+  const uint32_t num_inserts = 100;
+
+  std::vector<catalog::col_oid_t> col_oids;
+  col_oids.reserve(table_schema_.GetColumns().size());
+  for (const auto &col : table_schema_.GetColumns()) {
+    col_oids.push_back(col.Oid());
+  }
+  ProjectionMap projection_list_indices = sql_table_->ProjectionMapForOids(col_oids);
+
+  for (uint32_t i = 0; i < num_inserts; i++) {
+    for (uint32_t j = 0; j < i; j++) {
+      auto *const insert_txn = txn_manager_->BeginTransaction();
+      auto *const insert_redo =
+          insert_txn->StageWrite(CatalogTestUtil::TEST_DB_OID, CatalogTestUtil::TEST_TABLE_OID, tuple_initializer_);
+      auto *const insert_tuple = insert_redo->Delta();
+      *reinterpret_cast<int32_t *>(insert_tuple->AccessForceNotNull(projection_list_indices[catalog::col_oid_t(1)])) =
+          static_cast<int32_t>(i) * 2;
+      sql_table_->Insert(insert_txn, insert_redo);
+      txn_manager_->Commit(insert_txn, transaction::TransactionUtil::EmptyCallback, nullptr);
+    }
+  }
+
+  auto *const scan_txn = txn_manager_->BeginTransaction();
+
+  auto result_analyze =
+      Analyze(CatalogTestUtil::TEST_DB_OID, CatalogTestUtil::TEST_TABLE_OID, table_schema_, sql_table_, scan_txn, 10);
+
+  txn_manager_->Commit(scan_txn, transaction::TransactionUtil::EmptyCallback, nullptr);
+
+  std::vector<double> result_most_common_values{198, 196, 194, 192, 190, 188, 186, 184, 182, 180};
+  std::vector<double> result_most_common_frequencies{99, 98, 97, 96, 95, 94, 93, 92, 91, 90};
+  std::vector<double> result_histogram_bounds{2, 198};
```

Comment:
Instead of hardcoding these, can you automatically derive them when you insert the tuples above? This is going to be a pain to maintain in the future.